tidy3d/components/monitor.py

```diff
@@ -667,6 +667,69 @@ def storage_size(self, num_cells: int, tmesh: ArrayLike[float, 1]) -> int:
         return BYTES_COMPLEX * len(self.ux) * len(self.uy) * len(self.freqs) * 4
 
 
+class DiffractionEfficiencyMonitor(FieldMonitor, PlanarMonitor):
+    """:class:`Monitor` that uses a 2D Fourier transform to compute the
+    diffraction efficiency for given (or all) diffraction orders.
+
+    Example
+    -------
+    >>> monitor = DiffractionEfficiencyMonitor(
+    ...     center=(1,2,3),
+    ...     size=(5,5,0),
+    ...     freqs=[250e12, 300e12],
+    ...     name='diffraction_efficiency_monitor',
+    ...     normal_dir='+',
+    ...     orders_x=[0, 1, 2],
+    ...     orders_y=[0],
+    ...     )
+    """
+
+    normal_dir: Direction = pydantic.Field(
+        "+",
+        title="Normal vector orientation",
+        description="Direction of the surface monitor's normal vector w.r.t. "
+        "the positive x, y or z unit vectors. Must be one of ``'+'`` or ``'-'``. "
+        "Defaults to ``'+'`` if not provided.",
+    )
+
+    orders_x: Tuple[int, ...] = pydantic.Field(
+        None,
+        title="Diffraction orders along x",
+        description="Diffraction orders along x for which efficiency should be computed. "
+        "If ``None``, all orders between -1000 and 1000 will be considered.",
+    )
+
+    orders_y: Tuple[int, ...] = pydantic.Field(
+        None,
+        title="Diffraction orders along y",
+        description="Diffraction orders along y for which efficiency should be computed. "
+        "If ``None``, all orders between -1000 and 1000 will be considered.",
+    )
+
+    @pydantic.root_validator()
+    def default_diffraction_orders(cls, values):
+        """Set default diffraction orders if none are given."""
+        orders_x = values.get("orders_x")
+        orders_y = values.get("orders_y")
+
+        if orders_x is None:
+            values.update({"orders_x": list(range(-100, 100))})
```

Comment:
says 1000 in the docstring but 100 here. I think this number should be a constant defined at the top of the file. eg.  `MAX_ORDERS = 100`